source/structuring.rst

```diff
@@ -139,46 +141,190 @@ schema for a customer:
       }
     }
 
+.. note::
+
+    Even though the value of a ``$ref`` is a URI, it is not a network locator,
+    only an identifier. The schema may not actually exist at the address
+    specified. It is basically up to the validator implementation how external
+    schema URIs will be handled, but one should not assume the validator will
+    fetch network resources indicated in ``$ref`` values.
+
+Recursion
+`````````
+
+``$ref`` elements may be used to create recursive schemas that refer to themselves.
+For example, you might have a ``person`` schema that has an array of ``children``, each of which are also ``person`` instances.
+
+.. schema_example::
+
+    {
+      "$schema": "http://json-schema.org/draft-06/schema#",
+
+      "definitions": {
+        "person": {
+          "type": "object",
+          "properties": {
+            "name": { "type": "string" },
+            "children": {
+              "type": "array",
+    *          "items": { "$ref": "#/definitions/person" },
+              "default": []
+            }
+          }
+        }
+      },
+
+      "type": "object",
+
+      "properties": {
+        "person": { "$ref": "#/definitions/person" }
+      }
+    }
+    --
+    // A snippet of the British royal family tree
+    {
+      "person": {
+        "name": "Elizabeth",
+        "children": [
+          {
+            "name": "Charles",
+            "children": [
+              {
+                "name": "William",
+                "children": [
+                  { "name": "George" },
+                  { "name": "Charlotte" }
+                ]
+              },
+              {
+                "name": "Harry"
+              }
+            ]
+          }
+        ]
+      }
+    }
+
+However, a loop of ``$ref`` schemas referring to one another could cause
+infinite recursion in the validator, and is explicitly disallowed.
```

Comment:
Is it worth expanding this to explain that loops are allowed, but this example shows an unresolveable $ref resolution loop, not a validation loop?